internal/pkg/overlay/overlay.go

```diff
@@ -102,40 +101,76 @@ func (this Overlay) IsDistributionOverlay() bool {
 	return path.Dir(this.Path()) == config.Get().Paths.DistributionOverlaydir()
 }
 
-/*
-Build all overlays for a node
-*/
-func BuildAllOverlays(nodes []node.Node) error {
-	for _, n := range nodes {
+func BuildAllOverlays(nodes []node.Node, allNodes []node.Node, workerCount int) error {
+	nodeChan := make(chan node.Node, len(nodes))
+	errChan := make(chan error, len(nodes)*2)
 
-		sysOverlays := n.SystemOverlay
-		wwlog.Info("Building system overlays for %s: [%s]", n.Id(), strings.Join(sysOverlays, ", "))
-		err := BuildOverlay(n, "system", sysOverlays)
-		if err != nil {
-			return fmt.Errorf("could not build system overlays %v for node %s: %w", sysOverlays, n.Id(), err)
-		}
-		runOverlays := n.RuntimeOverlay
-		wwlog.Info("Building runtime overlays for %s: [%s]", n.Id(), strings.Join(runOverlays, ", "))
-		err = BuildOverlay(n, "runtime", runOverlays)
-		if err != nil {
-			return fmt.Errorf("could not build runtime overlays %v for node %s: %w", runOverlays, n.Id(), err)
+	var wg sync.WaitGroup
+	worker := func() {
+		for n := range nodeChan {
+			wwlog.Info("Building system overlays for %s: [%s]", n.Id(), strings.Join(n.SystemOverlay, ", "))
+			if err := BuildOverlay(n, allNodes, "system", n.SystemOverlay); err != nil {
+				errChan <- fmt.Errorf("could not build system overlays %v for node %s: %w", n.SystemOverlay, n.Id(), err)
+			}
+
+			wwlog.Info("Building runtime overlays for %s: [%s]", n.Id(), strings.Join(n.RuntimeOverlay, ", "))
+			if err := BuildOverlay(n, allNodes, "runtime", n.RuntimeOverlay); err != nil {
+				errChan <- fmt.Errorf("could not build runtime overlays %v for node %s: %w", n.RuntimeOverlay, n.Id(), err)
+			}
 		}
+		wg.Done()
+	}
+
+	for i := 0; i < workerCount; i++ {
+		wg.Add(1)
+		go worker()
+	}
+	for _, n := range nodes {
+		nodeChan <- n
+	}
+	close(nodeChan)
+
+	wg.Wait()
+	close(errChan)
 
+	for err := range errChan {
+		return err
 	}
 	return nil
 }
 
-// TODO: Add an Overlay Delete for both sourcedir and image
+func BuildSpecificOverlays(nodes []node.Node, allNodes []node.Node, overlayNames []string, workerCount int) error {
```

Comment:
This is the parallel build as in #1604 so the commits are mangled together

Reply:
They aren't mangled together; this PR just has to come after #1604, because separating it from changes in #1604 was impractical / unnecessary. Now that #1604 is merged, this one can be considered separately.